exercise_3/main.cpp

```diff
@@ -118,20 +131,19 @@ void writeObjectsToFile( std::string fileName, std::vector<std::pair<int, std::s
 
 }
 
-
 int main(int argc, char* argv[])
 {
-	std::vector<std::pair<int, std::shared_ptr<IRecord>>> exams;
-	std::vector<std::pair<int, std::shared_ptr<IRecord>>> courses;
-	std::vector<std::pair<int, std::shared_ptr<IRecord>>> teachers;
-	std::vector<std::pair<int, std::shared_ptr<IRecord>>> students;
+	std::vector< std::unique_ptr<IRecord>> exams;
+	std::vector< std::unique_ptr<IRecord>> courses;
+	std::vector< std::unique_ptr<IRecord>> teachers;
+	std::vector< std::unique_ptr<IRecord>> students;
 
 	loadObjectsFromFile(argv[1],exams, courses, teachers,students);
```

Comment:
This function returns a result. I believe you'd want to handle `FileNotFound` and `UnknownRecordType` and return `EXIT_FAIL`.

Reply:
Could it be like :
LoadingResult result = loadObjectsFromFile(argv[1], exams, courses, teachers, students);

		if (result == LoadingResult::FileNotFound)
		{
			std::cerr << "File not found" << std::endl;
			return EXIT_FAILURE;
		}
		if (result == LoadingResult::UnknownRecordType)
		{
			std::cerr << "File not found" << std::endl;
			return EXIT_FAILURE;
		}

Or should I use try catch block?

Reply:
The if/else is fine.
You could use const auto: `const auto result = loadObjectsFromFile(...);`

Reply:
Ok, thank you ;)